src/components/PopupWithForm.js

```diff
@@ -15,19 +14,46 @@ export default class PopupWithForm extends Popup {
     });
     return inputValues;
   }
-  // close() {
-  //   this._popupForm.reset();
-  //   super.close();
-  // }
+
+  open() {
+    super.open();
+  }
 
   setEventListeners() {
     //event listeners
-    this._popupForm.addEventListener("submit", (e) => {
+    this._popup = this._popupForm =
+      this._popupElement.querySelector(".modal__form");
+    this._popup.addEventListener("submit", (e) => {
       e.preventDefault();
       const inputValues = this._getInputValues();
+      console.log(inputValues);
       this._handleFormSubmit(inputValues);
+
+      this._popupForm.reset();
       this.close();
     });
     super.setEventListeners();
   }
+
+  // setSubmitAction(action) {
+  //   this._handleFormSubmit = action;
+  // }
+
+  renderLoading(isLoading) {
+    this._modalButton = this._popupElement.querySelector(".modal__button");
+    if (isLoading) {
+      this._modalButton.textContent = "Saving...";
+    } else {
+      this._modalButton.textContent = "Save";
+    }
+  }
+
+  renderCreating(isLoading) {
+    this._modalButton = this._popupElement.querySelector(".modal__button");
+    if (isLoading) {
+      this._modalButton.textContent = "Creating...";
+    } else {
+      this._modalButton.textContent = "Create";
```

Comment:
You have universal `renderLoading`. Please, use it for any form loading not to duplicate almost the same code


`renderCreating` should be deleted.


If you want to change the button texts, then you need to pass the texts into the call


```js
  renderLoading(isLoading, buttonText="Save", loadingText="Saving...")  {


```

And then you can change it to `Create` and `Creating...` if you want

Reply:
You haven't corrected my remark https://github.com/MrWatson0531/se_project_aroundtheus/pull/1#discussion_r1577844426